packages/utils/src/carequality/patient-discovery-statistics.ts

```diff
@@ -0,0 +1,79 @@
+import * as dotenv from "dotenv";
+dotenv.config();
+// // keep that ^ on top
+import { patientDataFromResource } from "@metriport/core/external/carequality/pd/process-inbound-pd";
+import { MPIMetriportAPI } from "@metriport/core/mpi/patient-mpi-metriport-api";
+import { inboundPatientResourceSchema } from "@metriport/ihe-gateway-sdk/src/models/patient-discovery/patient-discovery-responses";
+import { QueryTypes, Sequelize } from "sequelize";
+import z from "zod";
+import { getEnvVarOrFail } from "../../../api/src/shared/config";
+
+const apiUrl = getEnvVarOrFail("API_URL");
+const mpi = new MPIMetriportAPI(apiUrl);
+
+const cxId = getEnvVarOrFail("CX_ID");
+const sqlDBCreds = getEnvVarOrFail("DB_CREDS");
+const dbCreds = JSON.parse(sqlDBCreds);
+const sequelize = new Sequelize(dbCreds.dbname, dbCreds.username, dbCreds.password, {
+  host: dbCreds.host,
+  port: dbCreds.port,
+  dialect: dbCreds.engine,
+  logging: false,
+});
+
+const rowWithDataSchema = z.object({
+  status: z.string(),
+  data: z
+    .object({
+      patientResource: inboundPatientResourceSchema.optional(),
+      timestamp: z.string(),
+    })
+    .optional(),
+});
+
+const patientId = "";
+
+async function main() {
+  const query = `
+  SELECT * FROM patient_discovery_result
+  WHERE data->>'cxId'='${cxId}'
+  and created_at > '2024-03-07'
+  `;
+  const patientFilter = patientId ? `and patient_id='${patientId}'` : ``;
+  const queryString = `${query} ${patientFilter};`;
+
+  const resp = await sequelize.query(queryString, {
+    type: QueryTypes.SELECT,
+  });
+
+  const numberOfRows = resp.length;
+  let numberOfMatches = 0;
+  let numberOfSuccesses = 0;
+  let numberOfPatients = 0;
+  await Promise.allSettled(
+    //eslint-disable-next-line @typescript-eslint/no-explicit-any
+    resp.map(async (r: any) => {
+      if (Object.keys(r.data.patientResource).length == 0) delete r.data.patientResource;
+      const row = rowWithDataSchema.parse(r);
+      if (row.status === "success") numberOfSuccesses++;
+
+      const patient = patientDataFromResource(row?.data?.patientResource);
+      if (!patient) return;
```

Comment:
If no patient we return but we already increased `numberOfSuccesses`. Is this correct/expected?

Reply:
Yes, that's expected. We're tracking 4 different things here. 
1) # PDs
2) # success responses
3) # parsed patients
4) # mpi matches

Reply:
I understand, but do we consider a response without patient a successful one?

Reply:
Yes, I believe so. The case 1 (success) channel assigns `patientResource` only if it exists - [here](https://github.com/metriport/metriport/blob/97c85c6b11a1663121da40d8fb64e2774cc1099c/packages/ihe-gateway/server/Channels/CareQuality%20Document%20Exchange%20-%20Outbound/XCPD%20ITI-55%20Processor/destinationConnector-XCPD%20Endpoint-responseTransformer-step-1-Case%201%3A%20One%20Match.js#L34-L36)

Reply:
But are we not disregarding those entries when we process them [here](https://github.com/metriport/metriport/pull/1747#discussion_r1518501620)? If we keep that, in theory not having a patient resource is a failure, right?

Reply:
Not necessarily. My thinking is as follows (again, theoretically, idk if it's actually like this in practice):
- We can get a success from a gateway, but not receive a `patientResource`, even tho we can do a DQ and receive documents as expected
- The stats will be able to tell us how many external matches we got and how many we parsed for MPI to do its thing. 

Thinking about it now, maybe I'll include how many `patientResources` we get to see if any gateways neglect sending them

Reply:
> include how many patientResources we get to see if any gateways neglect sending them

I think this would be good